oak_restricted_kernel/src/dice_attestation.rs

```diff
@@ -15,37 +15,25 @@
 //
 
 use coset::CborSerializable;
-use zerocopy::FromZeroes;
 
-/// Generates an ECA certificate for use by the application.
-fn generate_application_certificate(
-    kernel_signing_key: &p256::ecdsa::SigningKey,
-    kernel_cert_issuer: alloc::string::String,
-    application_verifying_key: &p256::ecdsa::VerifyingKey,
-    app_digest: &[u8],
-) -> coset::CoseSign1 {
-    let additional_claims = alloc::vec![(
-        coset::cwt::ClaimName::PrivateUse(oak_dice::cert::LAYER_2_CODE_MEASUREMENT_ID),
-        coset::cbor::value::Value::Bytes(app_digest.into()),
-    )];
-    oak_dice::cert::generate_signing_certificate(
-        kernel_signing_key,
-        kernel_cert_issuer,
-        application_verifying_key,
-        additional_claims,
-    )
-    .expect("couldn't generate signing certificate")
+fn certificate_to_owned_slice(
+    cert: coset::CoseSign1,
+) -> [u8; oak_dice::evidence::CERTIFICATE_SIZE] {
+    let vec = cert.to_vec().expect("couldn't serialize certificate");
+    let mut slice = [0; oak_dice::evidence::CERTIFICATE_SIZE];
+    slice[..vec.len()].copy_from_slice(&vec);
+    slice
 }
```

Comment:
functions could probably be shared with the orchestrator / stage1, but we can dedupe code through refactoring later.